### PR TITLE
Add logging to GUI application

### PR DESCRIPTION
**Pull request by cydanil**
This PR adds logging to the `ipassign` application.
To do so effectively, `NetworkInterface` now inherits from `QObject` in order to send logging messages over Qt Signals.

### CODE REVIEW

File: gui/mainwindow.py [hunk 1]
```diff
@@ -15,7 +15,6 @@ class MainWindow(QObject):
     configuration can be modified in the HostnameSettings or NetworkSettings
     windows.
     """
-
     def __init__(self, parent=None):
         super(MainWindow, self).__init__(parent=parent)
```
Comment (cydanil) [inline gui/mainwindow.py:19]: This should be 
```python
super().__init__(parent=parent)
```

Comment (cydanil) [reply gui/mainwindow.py:19]: And likewise in other classes

File: gui/configurations.py [hunk 2]
```diff
@@ -57,6 +57,7 @@ class HostnameWindow(QObject):
     """
 
     done = pyqtSignal()
+    log = pyqtSignal(str)
 
     def __init__(self, parent=None):
         super(HostnameWindow, self).__init__(parent=parent)
```
Comment (cydanil) [inline gui/configurations.py:63]:
```python
super().__init__(parent=parent)
```

**Comment by cydanil** [comment]
@kirienko, feeling blue? You too can review!

File: gui/configurations.py [hunk 3]
```diff
@@ -168,6 +171,7 @@ class NetworkWindow(QObject):
     """
 
     done = pyqtSignal()
+    log = pyqtSignal(str)
 
     def __init__(self, parent=None):
         super(NetworkWindow, self).__init__(parent=parent)
```
Comment (cydanil) [inline gui/configurations.py:177]:
```python
super().__init__(parent=parent)
```

File: gui/gui.md [hunk 4]
```diff
@@ -69,10 +63,20 @@ As with `HostnameWindow`, the hostname is validated against RFC 1123.
 
 If any field is invalid, `Apply` will be disabled.
 
+## Log Window
```
Comment (cydanil) [inline gui/gui.md:66]: Describe what sort of messages to expect in the logs.

Generally, this file needs a re-write

File: gui/main.py [hunk 5]
```diff
@@ -16,17 +17,23 @@ def main():
 
     app = QApplication(sys.argv)
 
-    w = QMainWindow()
-    main = MainWindow(w)
+    window = QMainWindow()
```
Comment (cydanil) [inline gui/main.py:20]: Although seemingly trivial, this file deserves more comments.

**Comment by cydanil** [comment]
Merged locally.

**Comment by kirienko** [comment]
Oh man, I tried to go into the details twice, and even read about a half of the MR, but it's too complicated. When I went here for the third time, you'd already closed it ¯\_(ツ)_/¯

Maybe the next time, if you pick a bit softer piece to bite :)

**Pull request closed without merging**